### PR TITLE
feat: add https support #546

### PR DESCRIPTION
**Pull request by maoo**
Implements https://github.com/finos/git-proxy/issues/546

### CODE REVIEW

**Comment by JamieSlome** [comment]
> ## [Codecov](https://app.codecov.io/gh/finos/git-proxy/pull/558?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos) Report
> Attention: Patch coverage is `44.44444%` with `10 lines` in your changes are missing coverage. Please review.
> 
> > Project coverage is 56.76%. Comparing base [(`297f7e7`)](https://app.codecov.io/gh/finos/git-proxy/commit/297f7e7c0c5ee5a4e12d5fb45ed6a44fef7df1fb?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos) to head [(`37fc948`)](https://app.codecov.io/gh/finos/git-proxy/pull/558?dropdown=coverage&src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos).
> 
> [Files](https://app.codecov.io/gh/finos/git-proxy/pull/558?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos)	Patch %	Lines
> [src/config/index.js](https://app.codecov.io/gh/finos/git-proxy/pull/558?src=pr&el=tree&filepath=src%2Fconfig%2Findex.js&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos#diff-c3JjL2NvbmZpZy9pbmRleC5qcw==)	37.50%	[10 Missing ⚠️ ](https://app.codecov.io/gh/finos/git-proxy/pull/558?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos)
> Additional details and impacted files
> [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/finos/git-proxy/pull/558?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos).

@maoo - can we get at least 80%? 🫶

**Comment by maoo** [comment]
> > ## [Codecov](https://app.codecov.io/gh/finos/git-proxy/pull/558?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos) Report
> > Attention: Patch coverage is `44.44444%` with `10 lines` in your changes are missing coverage. Please review.
> > > Project coverage is 56.76%. Comparing base [(`297f7e7`)](https://app.codecov.io/gh/finos/git-proxy/commit/297f7e7c0c5ee5a4e12d5fb45ed6a44fef7df1fb?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos) to head [(`37fc948`)](https://app.codecov.io/gh/finos/git-proxy/pull/558?dropdown=coverage&src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos).
> > 
> > 
> > [Files](https://app.codecov.io/gh/finos/git-proxy/pull/558?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos)	Patch %	Lines
> > [src/config/index.js](https://app.codecov.io/gh/finos/git-proxy/pull/558?src=pr&el=tree&filepath=src%2Fconfig%2Findex.js&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos#diff-c3JjL2NvbmZpZy9pbmRleC5qcw==)	37.50%	[10 Missing ⚠️ ](https://app.codecov.io/gh/finos/git-proxy/pull/558?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos)
> > Additional details and impacted files
> > [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/finos/git-proxy/pull/558?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=finos).
> 
> @maoo - can we get at least 80%? 🫶

how does 100% sound? 😄

File: certs/cert.pem
Comment (JamieSlome) [inline certs/cert.pem:1]: Forgive me @maoo - can you explain why this file is required? Is this common practice to include a hardcoded `.pem` file?

File: certs/key.pem
Comment (JamieSlome) [inline certs/key.pem:1]: Same point here, should we be included this in the codebase or is this rather a result of a command that should be run by the downstream developer and architect?

File: src/proxy/index.js [hunk 1]
```diff
@@ -29,9 +36,13 @@ const start = async () => {
...
 
-  proxyApp.listen(proxyHttpPort, () => {
-    console.log(`Proxy Listening on ${proxyHttpPort}`);
+  http.createServer(options, proxyApp).listen(proxyHttpPort, () => {
+    console.log(`HTTP Proxy Listening on ${proxyHttpPort}`);
   });
+  https.createServer(options, proxyApp).listen(proxyHttpsPort, () => {
+    console.log(`HTTPS Proxy Listening on ${proxyHttpsPort}`);
+  });
+
```
Comment (JamieSlome) [inline src/proxy/index.js:45]: Can you explain this change? Why has a new dependency been introduced and why are two servers being created, both HTTP and HTTPS?

File: test/testConfig.test.js [hunk 2]
```diff
@@ -90,6 +92,19 @@ describe('user configuration', function () {
...
+      sslCertPemPath: "my-cert.pem"
+    };
+    fs.writeFileSync(tempUserFile, JSON.stringify(user));
+
+    const config = require('../src/config');
+
+    expect(config.getSSLKeyPath()).to.be.eql(user.sslKeyPemPath);
+    expect(config.getSSLCertPath()).to.be.eql(user.sslCertPemPath);
+  });
+
```
Comment (JamieSlome) [inline test/testConfig.test.js:107]: Can you confirm? This confirms that some defined key and certificate paths are loaded correctly into config?

**Review by JamieSlome — COMMENTED** [review]
Thanks for a wonderful PR, @maoo. I've left a couple of comments to help enlighten my unenlightened mind on HTTP/HTTPs. Appreciate you as ever! 👍 ❤️

**Review by JamieSlome — APPROVED** [review]
LGTM! 🍰

**Pull request merged**